### ISSUE TITLE
Cannot use ansible-vault encrypt together with vault_identity_list in ~/.ansible.cfg

### ISSUE THREAD

**Issue by hjudt** [ansible/ansible#30878]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible-vault encrypt

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = /home/hjudt/.ansible.cfg
  configured module search path = ['/home/hjudt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib64/python3.5/site-packages/ansible
  executable location = /usr/lib/python-exec/python3.5/ansible
  python version = 3.5.3 (default, Sep 20 2017, 22:40:40) [GCC 6.4.0]
```

##### CONFIGURATION
ANSIBLE_SSH_PIPELINING(/home/user/.ansible.cfg) = True
DEFAULT_VAULT_IDENTITY_LIST(/home/user/.ansible.cfg) = ['prod@/home/user/ansible/vault/prod', 'test@/home/user/ansible/vault/test', 'dev@/home/user/ansible/vault/dev', 'staging@/home/user/ansible/vault/staging']

##### OS / ENVIRONMENT
Linux/x64 gentoo

##### SUMMARY
When using vault_identity_list in ~/.ansible.cfg causes ansible-vault encrypt refusing to do any work, complaining "ERROR! Only one --vault-id can be used for encryption".

##### STEPS TO REPRODUCE
1. add setting (see example above) to ~/.ansible.cfg.
2a. try to use ansible-vault encrypt --vault-id <id>@passwordfile file-to-encrypt and see it fail
2b. try to use ansible-vault encrypt file-to-encrypt and see it fail

But: Decrypting works flawlessly. Also, if you remove the line in the config and use the --vault-id command line parameter, encryption works without any issues.

##### EXPECTED RESULTS
File to be encrypted.

##### ACTUAL RESULTS
Ansible-vault encrypt complains: "ERROR! Only one --vault-id can be used for encryption"

**Comment by alikins** [comment]
Currently encrypt only works for one vault-id, so multiple entries in vault_identity_list (or multiple --vault-id/--ask-vault-pass/--vault-password-file cli args) will cause that error. 

I have a few ideas on how to fix:

1) add some config item like 'vault_encrypt_id' to specify which vault-id to use for encryption by default

2) add a --encrypt-vault-id cli arg to specify which vault-id to use for encrypt

   So something `ansible-vault encrypt --vault-id prod@prod_pass --vault-id dev@dev_pass --encrypt-vault-id dev some_file_to_encrypt.yml` would work. 

3) Just default to one of the passwords from the vault ids list

   Currently unsure if the first item or the last item makes more sense. On decrypt, they are all tried in order
   (so for your example config the order would be 'prod', 'test', 'dev', 'staging')

  But I'm not sure if picking 'prod' or 'staging' would be correct for encrypt. And using one the user didn't expect to encrypt seems bad.

I'm leaning towards 1 and 2 at the moment, but any suggestions would be appreciated.

**Comment by alikins** [comment]
There is some related discussion in the comments for https://github.com/ansible/ansible/issues/30491

**Comment by hjudt** [comment]
ad 1. It is a nice-to-have as additional feature, yet it will not solve this problem.

ad 2. This seems a solution that will work and is definitely needed to distinguish between passwords for different environments. Specifying only the first part of the vault id from the list in vault_identity_list (the `id' in id@password_file) would make usage of this feature much easier too; This would apply to all similar cli usages. It is a bit of an annoyance one always has to specify the whole id@password_file.

ad 3. This is maybe better though less explicit than 1.

**Comment by hjudt** [comment]
One other issue I've seen with this is that ansible-vault edit doesn't write the id to the header of the re-encrypted file; I will have to verify this again and will report back whether it is still the case in up-to-date git.

**Comment by aldanor** [comment]
Confirming this problem.

Just to add, this also fails if `vault_password_file` is set in `ansible.cfg` (same way as with `vault_identity_list`).

**Comment by chrisweeksnz** [comment]
Agreed with aldanor and the problem remains using ansible 2.4.2.0.

When using ansible.cfg to specify `vault_password_file`, then encrypting a file fails with `"ERROR! Only one --vault-id can be used for encryption"`

**Comment by nguyenkims** [comment]
@alikins do you know why this issue is closed? Is there a solution/workaround to avoid this error? It makes some sublime/vscode ansible-vault extension unusable actually.

**Comment by whater39** [comment]
The Bug for the "vault_identity_list" still appears to be an issue in 2.6.4 While this field is active, you can successfully decrypt and edit. Still can't encrypt.

The bug for "vault_password_file" seems to be resolved in 2.6.4 though. I'm able to encrypt/decrypt files while "vault_password_file" has a value in it.
